Compilation error on OpenBSD/clang

Since https://github.com/WebAssembly/wabt/commit/4c3189d1df4a4062fb2705182a525fd8c23d6af2, I get the following error:
```
[  1%] Building CXX object CMakeFiles/wabt.dir/src/apply-names.cc.o
In file included from /tmp/wabt/src/apply-names.cc:17:
In file included from /tmp/wabt/src/apply-names.h:20:
In file included from /tmp/wabt/src/common.h:37:
In file included from /tmp/wabt/src/string-view.h:23:
In file included from /usr/include/c++/v1/ostream:140:
In file included from /usr/include/c++/v1/locale:208:
/usr/include/c++/v1/__bsd_locale_fallbacks.h:123:17: error: use of undeclared identifier 'vasprintf'; did you mean 'vsprintf'?
    int __res = vasprintf(__s, __format, __va);
```

It seems like the problem is cause by `-D_POSIX_C_SOURCE=200809L`.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -133,9 +133,11 @@ else ()
   # definition, and some libcs (e.g. glibc2.17 and earlier) follow that.
   add_definitions(-D__STDC_LIMIT_MACROS=1 -D__STDC_FORMAT_MACROS=1)
 
-  # On MINGW, _POSIX_C_SOURCE is needed to ensure we use mingw printf
-  # instead of the VC runtime one.
-  add_definitions(-D_POSIX_C_SOURCE=200809L)
+  if (MINGW OR CYGWIN)
+    # On MINGW, _POSIX_C_SOURCE is needed to ensure we use mingw printf
+    # instead of the VC runtime one.
+    add_definitions(-D_POSIX_C_SOURCE=200809L)
+  endif()
 
   if (COMPILER_IS_GNU)
     # disable -Wclobbered: it seems to be guessing incorrectly about a local

